opm/autodiff/FlowMain.hpp

```diff
@@ -252,9 +253,67 @@ namespace Opm
 #endif
         }
 
+        /// checks cartesian adjacency of global indices g1 and g2
+        bool cartesianAdjacent(const Grid& grid, size_t g1, size_t g2) {
+            // we need cartDims from UgGridHelpers
+            using namespace UgGridHelpers;
 
+            auto diff = std::abs(g1 - g2);
```

Comment:
This is asking for overflow, and I'm extremely surprised if it works! Use int instead of size_t and you are safe. Also the auto type for diff is only obscuring this signed/unsigned problem, prefer int.